code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -734,18 +739,17 @@ pub mod pallet {
 					);
 
 					ensure!(
-						initial_reward_config.iter().all(|(_, reward_config)| {
+						initial_reward_config.iter().all(|(reward_asset_id, reward_config)| {
 							if reward_config.reward_rate.amount > T::Balance::zero() {
-								// If none zero reward, check that the slashed amount is greater
-								// than ED
+								// if the reward rate amount is non-zero, then ensure that the
+								// slashed amount is >= the exisistential deposit for this asset
 								lock.unlock_penalty
 									.left_from_one()
 									.mul(reward_config.reward_rate.amount) >=
-									existential_deposit
+									T::ExistentialDeposits::get(reward_asset_id)
```

Comment:
@PoisonPhang I believe you wrote this originally - can you confirm that this change makes sense? also cc @vimukthi-git here

Reply:
It makes sense, as discussed back then. Reward rates should allow always for the unlock penalty to be transferred if an early unlock happens.

Reply:
I was asking about the use of `T::ExistentialDeposits::get(reward_asset_id)` rather than the existential deposit of the pool asset. I believe the previous implementation was incorrect

Reply:
Yes, this makes sense +1